current_discussion/dataRepresentation.md

```diff
@@ -0,0 +1,81 @@
+# Data representation
+
+Follow-up on meeting of 30sep20 - Joachim Wuttke.
+
+## Common subset of YAML and HDF5, no attributes
+
+To keep things simple, our data model should only use
+components that are supported by YAML *and* HDF5. These
+are arrays (=lists), dictionaries (=maps=hashes) and
+scalars (numbers or strings). This precludes the use
+of HDF5 attributes.
+
+In standard ascii representation of HDF5, attributes
+are denoted by the prefix "@". In NeXuS, they are used
+most prominently to specifiy physical units. This is
+a sensible choice within HDF5, but not important enough
+to justify the complications it would cause for us.
+
+## Values and ranges with units
+
+This brings us to the question how to represent physical
+units. In the YAML community there seems to be consensus
+that this is best done by sticking with the hierarchical
+nature of the data format,
+```
+  wavelength: {value: 7.2, unit: angstrom}
+```
+
+Also specify that we want flow style here, not block style:
+```
+  wavelength:
+    value: 7.2
+    unit: angstrom
+```
+
+A related question: how to represent a range. Proposal:
+```
+  wavelength: {min:4, max:12, unit: angstrom}
```

Comment:
An alternative is:
```
wavelength_min: 4
wavelength_max: 12
```
And say that all definitions of wavelength/Q are required to be in Angstrom. I'm not fixed on Angstrom, nm is fine by me as well.

Reply:
How does CIF handle this?